### PR TITLE
feat: add invoice service

### PR DESCRIPTION
**Pull request by WebertySilveira**

### CODE REVIEW

File: app/Services/InvoiceService.php
Comment (kayobruno) [inline app/Services/InvoiceService.php:1]: Please, add unit tests for this class.
No test, no beer! 🍺

File: app/Services/InvoiceService.php [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+declare(strict_types=1);
+
+namespace App\Services;
+
+use App\Contracts\DataParam;
+use App\Repositories\InvoiceEloquentRepository;
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Pagination\LengthAwarePaginator;
+use phpDocumentor\Reflection\Types\Collection;
```
Comment (kayobruno) [inline app/Services/InvoiceService.php:11]: Are you sure if the import of this class is correct?

File: app/Services/InvoiceService.php [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    {
+        return $this->invoiceRepository->getAll();
+    }
+
+    public function paginate(): LengthAwarePaginator
+    {
+        return $this->invoiceRepository->paginate();
+    }
+
+    public function findById(string $id): Model
```
Comment (kayobruno) [inline app/Services/InvoiceService.php:29]: You can use the `Invoice` like return type.

File: app/Services/InvoiceService.php [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    {
+        return $this->invoiceRepository->findById($id);
+    }
+
+    public function save(DataParam $dataParam): Model
```
Comment (kayobruno) [inline app/Services/InvoiceService.php:34]: same as above

File: app/Services/InvoiceService.php [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    {
+        return $this->invoiceRepository->save($dataParam);
+    }
+
+    public function update(string $id, DataParam $dataParam): Model
```
Comment (kayobruno) [inline app/Services/InvoiceService.php:39]: same as above

Comment (kayobruno) [inline app/Services/InvoiceService.php:39]: You can use the `InvoiceData` like param type.

File: app/Services/InvoiceService.php (hunk 3 above)
Comment (kayobruno) [inline app/Services/InvoiceService.php:34]: You can use the `InvoiceData` like param type.

**Review by kayobruno — CHANGES_REQUESTED** [review]

**Comment by WebertySilveira** [comment]
It was necessary to combine service and repository as they depend on each other

**Pull request closed without merging**